src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -59,30 +58,40 @@ private RazorLanguageServer(ILanguageServer innerServer)
 
         public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output, Trace trace, Action<RazorLanguageServerBuilder> configure = null)
         {
-            Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
+            // Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
             Serializer.Instance.JsonSerializer.Converters.RegisterRazorConverters();
 
             // Custom ClientCapabilities deserializer to extract experimental capabilities
             Serializer.Instance.JsonSerializer.Converters.Add(ExtendableClientCapabilitiesJsonConverter.Instance);
 
             ILanguageServer server = null;
+            var logLevel = trace switch
+            {
+                Trace.Messages => LogLevel.Information,
+                Trace.Off => LogLevel.None,
+                Trace.Verbose => LogLevel.Trace,
+                _ => default,
+            };
+
             server = OmniSharp.Extensions.LanguageServer.Server.LanguageServer.PreInit(options =>
                 options
                     .WithInput(input)
                     .WithOutput(output)
+                    .WithContentModifiedSupport(false)
+                    .WithSerializer(Serializer.Instance)
                     .ConfigureLogging(builder => builder
-                        .AddLanguageServer(RazorLSPOptions.GetLogLevelForTrace(trace))
-                        .SetMinimumLevel(LogLevel.Trace)) // We set the minimum level here to "Trace" to ensure that other providers still get the opportunity to act on logs if they prefer.
-                    .OnInitialized(async (s, request, response) =>
+                        .SetMinimumLevel(RazorLSPOptions.GetLogLevelForTrace(trace))
+                        .AddLanguageProtocolLogging(logLevel))
```

Comment:
Is this the bit that does textDocument/notification's?

Reply:
I think so, but maybe @david-driscoll can confirm.

Reply:
With the creation of a strongly typed client, this is just a rename from `.AddLanguageServer` if I recall.

Are you asking about the feature that wires in the logger to send logs to the client?

Reply:
> Are you asking about the feature that wires in the logger to send logs to the client?

ya

Reply:
You ask that, it looks like that functionality was removed (by accident!). However the method for that is `AddDefaultLoggingProvider`.  I will fix the functionality however.